python/ray/tune/tests/test_searchers.py

```diff
@@ -520,32 +521,42 @@ def setUpClass(cls):
     def tearDownClass(cls):
         ray.shutdown()
 
+    def _on_trial_callbacks(self, searcher, trial_id):
+        result = {
+            TRAINING_ITERATION: 1,
+            self.metric_name: 1,
+            "config/a": 1.0,
+            "time_total_s": 1,
+        }
+        searcher.on_trial_result(trial_id, result)
+        searcher.on_trial_complete(trial_id, result)
+
     def _save(self, searcher):
         searcher.set_search_properties(
             metric=self.metric_name, mode="max", config=self.config
         )
 
         searcher.suggest("1")
         searcher.suggest("2")
-        searcher.on_trial_complete(
-            "1", {self.metric_name: 1, "config/a": 1.0, "time_total_s": 1}
-        )
+        searcher.suggest("not_completed")
+        self._on_trial_callbacks(searcher, "1")
 
         searcher.save(self.checkpoint_path)
 
     def _restore(self, searcher):
-        searcher.set_search_properties(
-            metric=self.metric_name, mode="max", config=self.config
-        )
+        # Restoration shouldn't require another call to `searcher.set_search_properties`
         searcher.restore(self.checkpoint_path)
 
-        searcher.on_trial_complete(
-            "2", {self.metric_name: 1, "config/a": 1.0, "time_total_s": 1}
-        )
+        self._on_trial_callbacks(searcher, "2")
         searcher.suggest("3")
-        searcher.on_trial_complete(
-            "3", {self.metric_name: 1, "config/a": 1.0, "time_total_s": 1}
-        )
+        self._on_trial_callbacks(searcher, "3")
+
+        # NOTE: Trial "not_completed" that was suggested before saving never completes
```

Comment:
Previously, `BOHB` and `HyperOptSearch` did not load their `_live_trial_mapping` state, so trials suggested before the restore would report results/complete without actually updating the underlying searcher (aka they wouldn't add more data to the bayesopt).